toolkit/tools/internal/network/network.go

```diff
@@ -61,3 +64,34 @@ func DownloadFile(url, dst string, caCerts *x509.CertPool, tlsCerts []tls.Certif
 
 	return
 }
+
+// CheckNetworkAccess checks whether the installer environment has network access
+func CheckNetworkAccess() (err error) {
```

Comment:
Going back to our discussion: that's not exactly what I had in mind. I was thinking about separating `err` indicating function failure from the no-failure result when there's no network. For instance:

```suggestion
func CheckNetworkAccess() (err error, hasNetworkAccess bool) {
```